### PR TITLE
Kubernetes support - Customer Managed Agents Blog

### PR DESCRIPTION
**Pull request by meagancojocar**

### CODE REVIEW

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+meta_desc: Introducing Kubernetes-native support for Customer Managed Agents in Pulumi Cloud for enhanced flexibility and control over your infrastructure deployments.
+meta_image: meta.png
+authors:
+    - levi-blackstone
+    - meagan-cojocar
+tags:
+    - releases
+---
+
+We are excited to announce the availability of Kubernetes-native support for Customer-Managed Agents for Pulumi Deployments, further enhancing the flexibility and control over your infrastructure deployments. This new feature allows you to self-host deployment agents within your Kubernetes environment, bringing the same power and flexibility of Pulumi-hosted deployments to isolated environments.
```
Comment (cleverguy25) [inline content/blog/customer-managed-agents-kubernetes/index.md:15]:
```suggestion
We are excited to announce the availability of Kubernetes-native support for Pulumi Deployment Customer-Managed Agents, further enhancing the flexibility and control over your infrastructure deployments. This new feature allows you to self-host deployment agents within your Kubernetes environment, bringing the same power and flexibility of Pulumi-hosted deployments to isolated environments.
```

The multiple for ... for... was hard to read.

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Customer Managed Agents: Flexibility and Control
```
Comment (cleverguy25) [inline content/blog/customer-managed-agents-kubernetes/index.md:17]: Meta comment, technically we supported k8s before, but it was not widely circulated and most k8s admins would reject as it require pod security elevation, do we want to talk to that?

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+### Key Features
+
+1. **Multiple Deployment Triggers**: Supports all Pulumi deployment triggers such as Drift Detection, Scheduled Deployments, click to deploy, REST API, Git push, Review Stacks, and remote Automation API.
+2. **OpenID Authentication**: Fetch tokens dynamically for increased security.
+3. **Flexible Configuration**: Configure through YAML files or environment variables, adaptable to various deployment environments.
+
+### Docker and Kubernetes Support
+
+By default, Customer-Managed Agents are deployed using Docker. However, with this new release, you can now leverage Kubernetes for running these agents, enabling you to deploy it  within your existing infrastructure.
```
Comment (cleverguy25) [inline content/blog/customer-managed-agents-kubernetes/index.md:34]:
```suggestion
By default, Customer-Managed Agents are deployed on Virtual Machines using Docker. However, with this new release, you can now leverage Kubernetes to run these agents, enabling you to deploy them within your existing infrastructure.  This integrates natively with Kubernetes pods and does not require docker in docker or privileged execution.
```

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+To use Kubernetes for running Customer-Managed Agents:
+
+1. **Install the Pulumi GitHub App**: Ensure it's integrated with your source control.
+2. **Set Up Deployment Pools**:
+   - Navigate to Deployment Runners under Organization Settings in the left hand navigation bar.
+   - Create a new pool by clicking 'Add new pool'
+   - You will get a new access token, copy and save the token.
+3. **Install Agents**:
+   - Follow the provided instructions to install agents within your Kubernetes environment.
```
Comment (cleverguy25) [inline content/blog/customer-managed-agents-kubernetes/index.md:46]: Which instructions, is this going to be a link?

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+4. **Configure Agents**:
+   - Use `pulumi-deployment-agent.yaml` for configuration.
```
Comment (cleverguy25) [inline content/blog/customer-managed-agents-kubernetes/index.md:48]: While the yaml will work, I think we have a pulumi program example for this.

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+- **Scalability**: Set up multiple agent pools and scale dynamically, supporting up to 150 concurrent deployments.
+- **Versatility**: Mix and match Pulumi-hosted and self-hosted deployments to suit different infrastructure needs.
+
+### Key Features
+
+1. **Multiple Deployment Triggers**: Supports all Pulumi deployment triggers such as Drift Detection, Scheduled Deployments, click to deploy, REST API, Git push, Review Stacks, and remote Automation API.
+2. **OpenID Authentication**: Fetch tokens dynamically for increased security.
+3. **Flexible Configuration**: Configure through YAML files or environment variables, adaptable to various deployment environments.
+
+### Docker and Kubernetes Support
```
Comment (cleverguy25) [inline content/blog/customer-managed-agents-kubernetes/index.md:32]: While I agree with reiterating the value proposition, this benefits/key features seems to bury the lead and push the actual change quite far down the page.

File: content/blog/customer-managed-agents-kubernetes/index.md (hunk 2 above)
Comment (meagancojocar) [reply content/blog/customer-managed-agents-kubernetes/index.md:17]: Not really as I think it makes it more exciting to focus on what is new, but let me know if you think its important context to include.

File: content/blog/customer-managed-agents-kubernetes/index.md (hunk 5 above)
Comment (meagancojocar) [reply content/blog/customer-managed-agents-kubernetes/index.md:48]: How can I get this?

**Comment by arunkumar611** [comment]
Two feedback from one
1. In the getting started step 1, I'd link to this docs page: https://www.pulumi.com/docs/using-pulumi/continuous-delivery/github-app/
2. "Customer Managed Agents: Flexibility and Control" - this felt like an intro which is important. Is "Key Features" also part of the intro, or is that for K8 support? Its unclear to me because of the Heading levels 
3. Assuming this header should be one level higher? "Docker and Kubernetes Support"

File: content/blog/customer-managed-agents-kubernetes/index.md (hunk 5 above)
Comment (cleverguy25) [reply content/blog/customer-managed-agents-kubernetes/index.md:48]: https://github.com/pulumi/customer-managed-deployment-agent/tree/main/kubernetes

File: content/blog/customer-managed-agents-kubernetes/index.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+## Customer Managed Agents: Flexibility and Control
+
+Customer Managed Agents, [announced earlier this year](/blog/customer-managed-deployment-agents-launch), allow you to self-host deployment agents, delivering the power and flexibility of Pulumi Deployments within your own infrastructure. Here’s how you can benefit:
+
+- **Host Anywhere**: Deploy Customer Managed Agents on any hardware or environment of your choice, including fully private VPCs.
+- **Compliance and Security**: Keep your cloud provider credentials within your private network, meeting stringent compliance requirements.
+- **Scalability**: Set up multiple agent pools and scale dynamically, supporting up to 150 concurrent deployments.
+- **Versatility**: Mix and match Pulumi-hosted and self-hosted deployments to suit different infrastructure needs.
+
+## Key Features of Pulumi Deployments
```
Comment (arunkumar611) [inline content/blog/customer-managed-agents-kubernetes/index.md:26]:
```suggestion
### Key Features of Pulumi Deployments

Customer-Managed agents support all the capabilities of [Pulumi Deployments](/docs/pulumi-cloud/deployments/) including:
```

**Review by arunkumar611 — APPROVED** [review]
Looks good

**Pull request merged**